### PR TITLE
Optimize the database queries

### PR DESCRIPTION
**Pull request by leofeyer**
This PR relates to #7450.

### CODE REVIEW

**Comment by aschempp** [comment]
Would it be useful to move `$time = time() - (time() % 60);` to something like `\Date::minute()`?

**Comment by leofeyer** [comment]
As discussed on Mumble on March, 26th, we should add a method `Date::floorToMinute()`:

```php
public static function floorToMinute($time = null)
{
    if ($time === null)
    {
        $time = time();
    }

    return $time - ($time % 60);
}
```

File: system/modules/comments/dca/tl_comments.php [hunk 1]
```diff
@@ -30,8 +30,9 @@
 			'keys' => array
 			(
 				'id' => 'primary',
-				'source' => 'index',
-				'parent' => 'index'
+				'source' => 'index', // FIXME: remove?
+				'parent' => 'index', // FIXME: remove?
+				'source,parent,published' => 'index'
```
Comment (leofeyer) [inline system/modules/comments/dca/tl_comments.php]: As discussed on Mumble on March, 26th, we should remove the `source` and `parent` field indexes and add a stand-alone index on the `published` field.

File: system/modules/comments/dca/tl_comments_notify.php [hunk 2]
```diff
@@ -26,10 +26,11 @@
 			'keys' => array
 			(
 				'id' => 'primary',
-				'source' => 'index',
-				'parent' => 'index',
-				'tokenConfirm' => 'index',
-				'tokenRemove' => 'index'
+				'source' => 'index', // FIXME: remove?
+				'parent' => 'index', // FIXME: remove?
+				'tokenConfirm' => 'index', // FIXME: remove?
```
Comment (leofeyer) [inline system/modules/comments/dca/tl_comments_notify.php]: As discussed on Mumble on March, 26th, we should remove the single indexes on `source`, `parent` and `tokenConfirm`.

File: system/modules/core/dca/tl_page.php [hunk 3]
```diff
@@ -43,9 +43,9 @@
 			'keys' => array
 			(
 				'id' => 'primary',
-				'pid' => 'index',
 				'alias' => 'index',
-				'type' => 'index'
+				'type' => 'index', // FIXME: remove?
```
Comment (leofeyer) [inline system/modules/core/dca/tl_page.php]: As discussed on Mumble on March, 26th, the single index on the `type` column can be removed.

**Comment by leofeyer** [comment]
Merged in 573b4b67a4d2c3a8f3fd28ace1a34392c1675ae6.

**Pull request merged**